src/main/webapp/components/workspaces/workspaces.js

```diff
@@ -59,6 +68,181 @@ const workspaceById = gql`
     }
   }
 `
+const queryAttributes = gql`
+  fragment QueryAttributes on MetacardAttributes {
+    title
+    cql
+    metacard_tags
+    metacard_type
+  }
+`
+
+const useCreateQuery = workspaceId => {
+  const [queryId, setQueryId] = React.useState('')
+  const mutation = gql`
+    mutation CreateQuery($attrs: MetacardAttributesInput!) {
+      createMetacard(attrs: $attrs) {
+        id: id
+        ...QueryAttributes
+      }
+    }
+    ${queryAttributes}
+  `
+
+  const saveMutation = gql`
+    mutation SaveWorkspace($id: ID!, $attributes: Json!) {
+      saveMetacardFromJson(id: $id, attributes: $attributes) {
+        ...WorkspaceAttributes
+      }
+    }
+    ${workspaceAttributes}
+  `
+
+  const [save] = useMutation(saveMutation)
+
+  const [create] = useMutation(mutation, {
+    onCompleted: data => {
+      save({
+        variables: {
+          id: workspaceId,
+          attributes: {
+            queries: [data.createMetacard.id],
+            metacard_type: 'workspace',
+          },
+        },
+        refetchQueries: ['WorkspaceById'],
+      }),
+        setQueryId(data.createMetacard.id)
+    },
+  })
+
+  return [create, queryId]
+}
+
+const TextSearchForm = props => {
+  const [textValue, setTextValue] = React.useState('')
+  const [title, setTitle] = React.useState('')
+  const [create, queryId] = useCreateQuery(props.workspaceId)
+  const { runQuery } = React.useContext(WorkspaceContext)
+
+  const onCreate = async () => {
+    const filter = {
+      property: 'anyText',
+      type: 'ILIKE',
+      value: textValue,
+    }
+
+    const cql = transformFilterToCQL(filter)
+    try {
+      await create({
+        variables: {
+          attrs: {
+            title: title,
+            cql: cql,
+            metacard_tags: ['query'],
+            metacard_type: 'metacard.query',
+          },
+        },
+      })
+    } catch (err) {
+      //eslint-disable-next-line
```

Comment:
If I'm not mistaken, console logs get stripped when the build is ran in production mode, so if that's the case you guys should remove the eslint `no-console` rule so you don't have to add these comments everywhere.

Reply:
I think it's nice to have these explicitly ignored because some people use console logs to debug and we don't have a strict review process, so I think random console logs would get in